precompiles/src/precompile_set.rs

```diff
@@ -304,13 +304,11 @@ impl<T: SelectorFilter> PrecompileChecks for CallableByPrecompile<T> {
 #[derive(PartialEq)]
 #[cfg_attr(feature = "std", derive(Debug))]
 pub enum AddressType {
-	/// The code stored at the address is less than 5 bytes, but not well known.
-	Unknown,
```

Comment:
Shouldn't this be marked as deprecated, instead of removed ?
I suppose that removing it might break existing precompile code / invariants.

Reply:
After putting some thought into it, I think it should be removed instead of deprecated. This code was originally on the moonbeam repository and was then moved frontier. In my opinion, the impact of removing this is very low, it is not used on moonbeam and the enum never implemented encoding methods.